backend/src/Edna/Upload/Service.hs

```diff
@@ -48,20 +45,11 @@ parseFile' = measurementsToSummary . fcMeasurements
 
 compoundNameToId :: Text -> Edna (Maybe (SqlId 'U.CompoundId))
 compoundNameToId compoundName =
-  fmap (fmap mkSqlId) . runSelectReturningOne' $ select $ do
-    compound <- all_ (esCompound ednaSchema)
-    guard_ (cName compound ==. val_ compoundName)
-    return $ cCompoundId compound
+  (SqlId . unSerial . cCompoundId) <<$>> LQ.getCompoundByName compoundName
```

Comment:
Since you are doing some refactoring, I propose to define `SqlSerial Word32 -> SqlId anything` helper somewhere.